Assets/_BForBoss/_LevelDesign/Managers/StateManager.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+
+namespace BForBoss
+{
+    public enum State
+    {
+        PreGame,
+        Play,
+        Pause,
+        Death,
+    }
+    
+    public class StateManager
```

Comment:
This is great, only thing missing, is an interface where you can getState and SetState().

So that you can inject a MockStateManager into any manager/component that needs it.